src/gicv3.rs

```diff
@@ -20,6 +20,39 @@ use core::{
 /// The offset in bytes from `RD_base` to `SGI_base`.
 const SGI_OFFSET: usize = 0x10000;
 
+enum BitOp {
+    Set,
+    Clear,
+}
+
+fn modify_bit(registers: *mut [u32], nth: usize, op: BitOp) {
+    let reg_num: usize = nth / 32;
+    assert!(reg_num < registers.len());
+
+    let bit_num: usize = nth % 32;
+    let bit_mask: u32 = 1 << bit_num;
+
+    unsafe {
```

Comment:
Please add a safety comment for this unsafe block, explaining why the unsafe operations within are safe. This will need to rely on the safety preconditions of `modify_bit`.

Reply:
I'm pretty sure raw slices still have bounds checks, so I don't think you even need the assert.